typescript/src/main.ts

```diff
@@ -45,7 +45,46 @@ function appendMetalTableRow(metal: metals.Metal, table: HTMLTableElement) {
     .getElementsByTagName("tbody")[0]
     .insertRow();
 
-  row.insertCell(-1).outerHTML = "<th>" + metal.symbol + "</th>";
+  const toggleButton = <HTMLInputElement>document.createElement("input");
+  const label = <HTMLLabelElement>document.createElement("label");
+  toggleButton.type = "checkbox";
+  toggleButton.classList.add("toggle");
+  toggleButton.id = "toggle_" + metal.idSuffix;
+  label.htmlFor = "toggle_" + metal.idSuffix;
+  const metalCell = <HTMLTableCellElement>document.createElement("th");
+
+  toggleButton.addEventListener("change", function () {
+    if (this.checked) {
+      (
+        document.getElementById(
+          "affinity_" + metal.idSuffix
+        ) as HTMLInputElement
+      ).disabled = true;
+      (
+        document.getElementById("bmc_" + metal.idSuffix) as HTMLInputElement
+      ).disabled = true;
+      metal.disableMetal();
+    } else {
+      (
+        document.getElementById(
+          "affinity_" + metal.idSuffix
+        ) as HTMLInputElement
+      ).disabled = false;
+      (
+        document.getElementById("bmc_" + metal.idSuffix) as HTMLInputElement
+      ).disabled = false;
+      metal.resetValues();
+    }
+    updateRow(metal);
```

Comment:
I think the code would be cleaner if you extracted this to a separate function.

Also I think you could avoid repeating code by doing something like:

```
       (
         document.getElementById("bmc_" + metal.idSuffix) as HTMLInputElement
       ).disabled = this.checked;
```

(and the same for `affinity`)

Reply:
Would I not still need to have the if{} for calling switchOffMetal()[which is what I updated the name of disableMetal() to] and resetValues() accordingly. I initially didn't ad the else{} part and was only working with the this.checked which caused a lot of problems. I guess I could compact it into one method but it would still need a bool variable or I could write 2 methods?

Reply:
Yes, you'd still need the if/else to call the right method on the metal, it's just saving a couple of lines of repeated code.